jalib/jalib.cpp

```diff
@@ -135,9 +137,12 @@ dup(int oldfd)
   REAL_FUNC_PASSTHROUGH(int, dup) (oldfd);
 }
 
-int
-dup2(int oldfd, int newfd)
-{
+int dup2(int oldfd, int newfd) {
+  struct rlimit file_limit;
+  getrlimit(RLIMIT_NOFILE, &file_limit);
+  JASSERT ( (unsigned int)newfd < file_limit.rlim_cur )
```

Comment:
This assert shouldn't be here. The caller is allowed to call with an invalid newfd. We should simply return the error code.

For the case, where DMTCP is trying to use a newfd beyond the current limits, please add checks at the call-site.

Also note that this wrapper is specific to jalib. The core and plugins won't end up here when calling dup2.

Reply:
Hi @karya0 ,
    I respectfully think the code should stand as is.  I've updated to add comments before the dup2 wrapper to explain my reasoning.
    In addition to those comments, I'll add that jalib should not be calling dup2 with a newfd beyond RLIMIT_NOFILE.  There are currently two dup2 call sites in jalib, and if either one uses a newfd that is too large, then this is a bug in DMTCP (since the call site is internal to DMTCP).  So, we should error out.

Reply:
I agree with @gc00 here.